debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/BaseChangeRecordEmitter.java

```diff
@@ -119,9 +119,12 @@ protected void emitUpdateAsPrimaryKeyChangeRecord(Receiver receiver, TableSchema
     private List<Column> getReselectColumns(Struct newValue) {
         final List<Column> reselectColumns = new ArrayList<>();
```

Comment:
My comment is not related to this fix, but it seems to me that here is no need to create a new `ArrayList` if there are no CLOB/BLOB columns in the table. This can be easily avoided by adding the following check at the beginning:
```
final List<Column> lobColumns = schema.getLobColumnsForTable(table.id());
if (lobColumns.isEmpty()) {
    return Collections.emptyList();
}
final List<Column> reselectColumns = new ArrayList<>();
for (Column column : lobColumns) {
    ...
```

Reply:
Great catch, ya this is still in the works as I need to add a test case, but I'll adjust this too; :+1: